ci: fix commitlint problem

Madhu-1 opened the pull request:
Still seeing the issue of the commitlint as below

fatal: unsafe repository ('/go/src/github.com/ceph/ceph-csi' is owned by someone else) To add an exception for this directory,
call:

git config --global --add safe.directory /go/src/github.com/ceph/ceph-csi

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

This is not fixed yet happening at https://github.com/ceph/ceph-csi/runs/6345425683?check_suite_focus=true#step:3:1419


Note:
```
Setting this in Dockerfile which is used for the tests instead of Makefile as the issue is not seen in the local environment and we run tests inside the container in the CI.
```

Madhu-1 committed:
ci: fix commitlint problem

Still seeing the issue of the commitlint
as below

fatal: unsafe repository
('/go/src/github.com/ceph/ceph-csi'
is owned by someone else)
To add an exception for this directory,
call:

git config --global --add safe.directory \
/go/src/github.com/ceph/ceph-csi
Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Rakshith-R reviewed (changes requested):
Let's add it within the script

https://github.com/ceph/ceph-csi/blob/9533889b643f8274e8a0aae5583b022ddb1f5ee3/Makefile#L153

The step needs to be executed within the container
https://github.com/ceph/ceph-csi/runs/6346056638?check_suite_focus=true

Madhu-1 commented:
> Let's add it within the script
> 
> https://github.com/ceph/ceph-csi/blob/9533889b643f8274e8a0aae5583b022ddb1f5ee3/Makefile#L153

Am still testing this one. I don't see the problem in the local setup. if adding to a workflow does not help. will add it there as the last option.

Madhu-1 committed:
ci: remove set-safe-directory from commitlint

Removed set-safe-directory option from the
commitlint.yaml as its not working as expected.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
@mergifyio backport release-v3.6
